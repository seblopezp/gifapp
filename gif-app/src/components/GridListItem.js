import React from "react";

export const GridListItem = ({ id, title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
