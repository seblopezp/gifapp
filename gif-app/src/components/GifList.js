import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GridListItem } from "../components/GridListItem";
export const GifList = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}
      <div className="card-grid">
        {images.map((img) => (
          <GridListItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
