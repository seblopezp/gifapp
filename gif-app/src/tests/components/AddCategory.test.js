import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import "@testing-library/jest-dom";
import React from "react";

describe("Pruebas en <AddCategory />", () => {
  const setCategories = jest.fn();

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";
    input.simulate("change", { target: { value } });
  });

  test("No debe de postear la informaciÃ³n con el submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe llamar al setCategories y limpiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
