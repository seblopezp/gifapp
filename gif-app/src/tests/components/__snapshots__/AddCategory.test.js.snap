// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input is-rounded"
    onChange={[Function]}
    placeholder="Ej: Studio Ghibli"
    type="text"
    value=""
  />
</form>
`;
